AssetRipper.Library/Exporters/Textures/YamlSpriteExporter.cs

```diff
@@ -0,0 +1,61 @@
+using AssetRipper.Core.Classes.Misc;
+using AssetRipper.Core.Interfaces;
+using AssetRipper.Core.Parser.Files.SerializedFiles;
+using AssetRipper.Core.Project.Collections;
+using AssetRipper.Core.Project.Exporters;
+using AssetRipper.Library.Configuration;
+using AssetRipper.SourceGenerated.Classes.ClassID_213;
+using AssetRipper.SourceGenerated.Classes.ClassID_687078895;
+
+namespace AssetRipper.Library.Exporters.Textures
+{
+	/// <summary>
+	/// Exports sprites as YAML assets.
+	/// </summary>
+	public sealed class YamlSpriteExporter : YamlExporterBase
+	{
+		private SpriteExportMode SpriteExportMode { get; }
+
+		public YamlSpriteExporter(LibraryConfiguration configuration)
+		{
+			SpriteExportMode = configuration.SpriteExportMode;
+		}
+
+		public override bool IsHandle(IUnityObjectBase asset) =>
+			SpriteExportMode == SpriteExportMode.Yaml && asset is ISprite or ISpriteAtlas;
+
+		public override IExportCollection CreateCollection(VirtualSerializedFile virtualFile, IUnityObjectBase asset)
+		{
+			if (asset is ISpriteAtlas)
+			{
+				return new EmptyExportCollection();
+			}
+			
+			// If a SpriteAtlas reference is serialized into this sprite,
+			// we must recover the m_RD field of the sprite from the SpriteAtlas.
+			ISprite sprite = asset as ISprite;
+			if (sprite.SpriteAtlas_C213.FindAsset(sprite.SerializedFile) is ISpriteAtlas atlas)
+			{
+				var spriteData = atlas.RenderDataMap_C687078895[sprite.RenderDataKey_C213];
+				var m_RD = sprite.RD_C213;
```

Comment:
I don't use `var`. I think my editor config makes this display as a warning. If not I need to change that.

Reply:
fixed